### PR TITLE
matinfo: add support for viewing ESSL1 code

### PR DESCRIPTION
**Pull request by elizagamedev**

### CODE REVIEW

File: libs/matdbg/src/ShaderInfo.cpp [hunk 1]
```diff
@@ -94,12 +94,12 @@ bool getMetalShaderInfo(const ChunkContainer& container, ShaderInfo* info) {
     return true;
 }
 
-bool getGlShaderInfo(const ChunkContainer& container, ShaderInfo* info) {
-    if (!container.hasChunk(ChunkType::MaterialGlsl)) {
+bool getGlShaderInfo(const ChunkContainer& container, ShaderInfo* info, ChunkType chunkType) {
```
Comment (poweifeng) [inline libs/matdbg/src/ShaderInfo.cpp:97]: Just roughly looking the three methods: `getGlShaderInfo` `getMetalShaderInfo`, `getVkShaderInfo`, looks like they are all the same except for the chunktype?  Maybe consolidate?

File: libs/matdbg/src/TextWriter.cpp [hunk 2]
```diff
@@ -402,13 +402,23 @@ static void printShaderInfo(ostream& text, const vector<ShaderInfo>& info,
     text << endl;
 }
 
-static bool printGlslInfo(ostream& text, const ChunkContainer& container) {
+static bool printGlslInfo(ostream& text, const ChunkContainer& container, ChunkType chunkType) {
     vector<ShaderInfo> info;
     info.resize(getShaderCount(container, ChunkType::MaterialGlsl));
-    if (!getGlShaderInfo(container, info.data())) {
+    if (!getGlShaderInfo(container, info.data(), ChunkType::MaterialGlsl)) {
```
Comment (poweifeng) [inline libs/matdbg/src/TextWriter.cpp:408]: So it's not `chunkType` here and the line above?  

Also, maybe now that we're passing chunkType to `get*ShaderInfo`, maybe we can further consolidate these methods?  (If the `get*ShaderInfo` can be consolidated into one).

**Review by pixelflinger — APPROVED** [review]

File: libs/matdbg/src/ShaderInfo.cpp (hunk 1 above)
Comment (elizagamedev) [reply libs/matdbg/src/ShaderInfo.cpp]: Done.

File: libs/matdbg/src/TextWriter.cpp (hunk 2 above)
Comment (elizagamedev) [reply libs/matdbg/src/TextWriter.cpp]: Oops, missed this.

**Review by bejado — APPROVED** [review]

**Review by poweifeng — APPROVED** [review]

**Pull request merged**